packages/components/src/List/Toolbar/ColumnChooser/hooks/columnChooserManager.hook.js

```diff
@@ -0,0 +1,150 @@
+import { useState } from 'react';
+import cloneDeep from 'lodash/cloneDeep';
+import { compareOrder } from '../service';
+
+export function organiseColumns(columns) {
+	return columns.sort(compareOrder).map((column, index) => {
+		return { ...column, order: index + 1 };
+	});
+}
+
+export function updateEditedColumns(state) {
+	return { ...state, editedColumns: organiseColumns(state.editedColumns) };
+}
+
+export function isValueCorrect(value, collectionLength) {
+	return Number.isInteger(value) && value <= collectionLength;
+}
+
+export function getOrderItem(order, index, length) {
+	if (index === length - 1) {
+		return order - 1;
+	}
+	return order + 1;
+}
+
+export function matchOrder(value) {
+	return function match({ order }) {
+		return order === value;
+	};
+}
+
+export function changeColumnAttribute(key) {
+	return function setAttribut(value, index) {
+		return function setColumn(state) {
+			const newColumns = state.editedColumns;
+			newColumns[index][key] = value;
+			return { ...state, editedColumns: newColumns };
+		};
+	};
+}
+
+const updateAttributeVisiblity = changeColumnAttribute('hidden');
+const updateAttributeOrder = changeColumnAttribute('order');
+
+export function useColumnChooserManager(columns, customSubmit) {
+	const [state, setState] = useState({
```

Comment:
Here i ask myself having too :
```
const [editedColumns, setEditedColumn] = useState(organiseColumns(columns)
const [selectAll, setSelectAll] = useState(false)
```
I feel the useState api more natural when having each parameter aside ( but sure if we need to update both in one time, it could be weird to call both setter 🤔 )

Reply:
I think in a custom hooks that will create verbosity not needed.

Reply:
const [editedColumns, setEditedColumn] = useState(organiseColumns(columns)
const [selectAll, setSelectAll] = useState(false)
I think it's great in a component not a custom hook.